contract/proxy-oracle/soroban/scripts/budget_report.py

```
@@ -0,0 +1,207 @@
+    all_ok = True
+
+    # --- Runtime tests ---
+    lines.append("## Runtime Scenarios (templar-proxy-oracle-soroban-contract)")
+    print("Running runtime tests...", flush=True)
+    runtime_ok, runtime_output = run_cargo_test(
+        manifest_root, "templar-proxy-oracle-soroban-contract"
+    )
+    runtime_results = extract_test_results(runtime_output)
+
+    if not runtime_ok and not runtime_results:
+        lines.append(f"  cargo test FAILED to run:")
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)